kernel/tests/read.rs

```diff
@@ -695,6 +699,136 @@ fn predicate_on_number() -> Result<(), Box<dyn std::error::Error>> {
+///             expr: UnaryOperation {
+///                 op: IsNull,
+///                 expr: Column(
+///                     "metaData.id",
+///                 ),
+///             },
+///         },
+///         UnaryOperation {
+///             op: Not,
+///             expr: UnaryOperation {
+///                 op: IsNull,
+///                 expr: Column(
+///                     "protocol.minReaderVersion",
+///                 ),
+///             },
+///         },
+///     ],
+/// }) = [0])
+/// with_row_group_filter(VariadicOperation {
+///     op: Or,
+///     exprs: [
+///         UnaryOperation {
+///             op: Not,
+///             expr: UnaryOperation {
+///                 op: IsNull,
+///                 expr: Column(
+///                     "metaData.id",
+///                 ),
+///             },
+///         },
+///         UnaryOperation {
+///             op: Not,
+///             expr: UnaryOperation {
+///                 op: IsNull,
+///                 expr: Column(
+///                     "protocol.minReaderVersion",
+///                 ),
+///             },
+///         },
+///     ],
+/// }) = [])
+/// with_row_group_filter(VariadicOperation {
+///     op: Or,
+///     exprs: [
+///         UnaryOperation {
+///             op: Not,
+///             expr: UnaryOperation {
+///                 op: IsNull,
+///                 expr: Column(
+///                     "metaData.id",
+///                 ),
+///             },
+///         },
+///         UnaryOperation {
+///             op: Not,
+///             expr: UnaryOperation {
+///                 op: IsNull,
+///                 expr: Column(
+///                     "protocol.minReaderVersion",
+///                 ),
+///             },
+///         },
+///     ],
+/// }) = [0])
+/// with_row_group_filter(BinaryOperation {
+///     op: LessThan,
+///     left: Column(
+///         "numeric.ints.int32",
+///     ),
+///     right: Literal(
+///         Integer(
+///             1000,
+///         ),
+///     ),
+/// }) = [])
+/// ```
+#[test]
+fn parquet_predicate_pushdown() -> Result<(), Box<dyn std::error::Error>> {
+    let expected_none: Vec<_> = ["+------+", "| bool |", "+------+", "+------+"]
+        .iter()
+        .map(|s| s.to_string())
+        .collect();
+    let expected_all: Vec<_> = [
+        "+-------+",
+        "| bool  |",
+        "+-------+",
+        "|       |",
+        "|       |",
+        "|       |",
+        "| false |",
+        "| true  |",
+        "+-------+",
+    ]
+    .iter()
+    .map(|s| s.to_string())
+    .collect();
+    let cases = vec![
+        (
+            Expression::column("numeric.ints.int32").lt(Expression::literal(1000i32)),
+            expected_none,
+        ),
+        (
+            Expression::column("numeric.ints.int32").gt(Expression::literal(1000i32)),
+            expected_all,
+        ),
+    ];
+    for (expr, expected) in cases.into_iter() {
+        read_table_data(
```

Comment:
you can use `read_table_data_str` and pass a `Vec<&str>` as the expected output so you don't have to map/to_string/collect above. Just a little cleaner since so many tests want to do it this way.